api/api.go

```diff
@@ -242,6 +255,55 @@ func OrbPublish(ctx context.Context, logger *logger.Logger,
 	return &response.PublishOrb.PublishOrbResponse, err
 }
 
+// OrbPublishID publishes a new version of an orb by id
+func OrbPublishID(ctx context.Context, logger *logger.Logger,
+	configPath string, orbID string, orbVersion string) (*PublishOrbResponse, error) {
```

Comment:
Not directly related to this PR, but I see Logging as a cross-cutting concern, so I generally don't like passing it as an argument. You could pull one from the context (I can send you a link to an example) if you don't want to use a global instance.

Reply:
I'm willing to accept that passing around a logger to every function is silly, and like to refactor that but think it's better to save that for another PR.

If you have any ideas or examples on how we should do that, I'm all ears! :bow:

Reply:
This is what I am planning on using in some projects: https://gist.github.com/dnephin/4b8106cf5b09494d1caecdf8f74ce88e